Yosemite/Yosemite/Stores/CouponStore.swift

```diff
@@ -50,10 +50,16 @@ public final class CouponStore: Store {
 
         switch action {
         case .synchronizeCoupons(let siteID, let pageNumber, let pageSize, let onCompletion):
-            methods.synchronizeCoupons(siteID: siteID,
-                                       pageNumber: pageNumber,
-                                       pageSize: pageSize,
-                                       onCompletion: onCompletion)
+            Task {
+                do {
+                    let hasMorePages = try await methods.synchronizeCoupons(siteID: siteID,
+                                                                            pageNumber: pageNumber,
+                                                                            pageSize: pageSize)
+                    onCompletion(.success(hasMorePages))
+                } catch {
+                    onCompletion(.failure(error))
+                }
+            }
```

Comment:
I'm not against this, but caution that changing the action handling in this way might make testing harder. You don't have an async chain when you dispatch this action, so you're reliant on the `onCompletion` handler being called and maybe using a `waitFor`. Those haven't played well with Tasks/checkedContinuations in the past.

Do you think it would be possible/safer to contain the async changes within `methods`? Yes, ideally the old method would call the new async one so there's still a touch of that issue, but you wouldn't neccesarily need that to work for the tests to pass.

Reply:
PS I'm seeing some issues running tests locally due to threading, things falling off the main thread potentially in the `synchronizeCoupons` Task. Perhaps it just wants to be MainActor, ideally from the tests or calling its completion handlers on main